modules/nlu/src/views/full/intents/UtterancesEditor.tsx

```diff
@@ -138,11 +139,29 @@ export class UtterancesEditor extends React.Component<Props> {
           show={this.state.showSlotMenu}
           onSlotClicked={this.tag.bind(this, editor)}
         />
-        <div className={style.utterances}>{children}</div>
+        <div className={style.utterances} onCopy={this.onCopy}>{children}</div>
       </div>
     )
   }
 
+  onCopy = event => {
+    const selection = document.getSelection().toString()
+    let lines: string[]
+    if (!selection.length) {
+      // Selected the whole component, we put all utterances in the clipboard
+      lines = valueToUtterances(this.state.value)
+    } else {
+      // Partial selection, we remove the heading numbers and empty lines
+      lines = selection.split('\n').map(txt =>
+        txt.replace(/^\d{1,4}$/, '')
```

Comment:
up to 9999 utterances ? + what if an utterance starts with a number, do you trim it ?

Reply:
can't really imagine more than 10k utterances :-p

it has to start and finish* with a number, so the utterance would have to be just a number.  but yeah, it would be filtered out.